cypress/integration/common/deal-wizard.ts

```diff
@@ -0,0 +1,91 @@
+import { Then, When } from "@badeball/cypress-cucumber-preprocessor/methods";
+
+Then("I am presented the option to choose a partner", () => {
+  cy.url().should("match", /(initiate\/token-swap$)/)
+})
+
+When("I go to previous step", () => {
+  cy.get("[data-test='wizard-previous-button']").click();
+})
+
+When("I try to proceed to next step", () => {
+  cy.get("[data-test='wizard-proceed-button']").click();
+})
+
+When("I try to navigate to proposal stage via stepper", () => {
+  cy.get("[data-test='wizard-manager-stepper']").within(() => {
+    cy.contains("[data-test='pstepper-step']", "Proposal").click();
+  });
+})
+
+When("I try to navigate to proposal lead stage via stepper", () => {
+  cy.get("[data-test='wizard-manager-stepper']").within(() => {
+    cy.contains("[data-test='pstepper-step']", "Lead Details").click();
+  });
+})
+
+When("I try to navigate to primary dao stage via stepper", () => {
+  cy.get("[data-test='wizard-manager-stepper']").within(() => {
+    cy.contains("[data-test='pstepper-step']", "Primary DAO").click();
+  });
+})
+
+When("I try to navigate to partner dao stage via stepper", () => {
+  cy.get("[data-test='wizard-manager-stepper']").within(() => {
+    cy.contains("[data-test='pstepper-step']", "Partner DAO").click();
+  });
+})
+
+When("I fill in proposal title correctly", () => {
+  cy.get("[data-test='proposal-title-field']").within(() => {
+    cy.get('input').type('Test proposal')
+  })
```

Comment:
You write "correctly", but don't verify correctness here, So either
1. remove "correctly" in description
2. add correctness

Reply:
What I meant is that it's filled in correctly according to the business logic. I'm not sure if that implies checking for error messages in the same step but I don't know I'm not really into linguistics :D  
I can change it to "I fill in proposal title"

Reply:
If you just use this step in the Scenario

```feature
  Scenario: I proceed from proposal stage after filling required fields correctly
    When I fill in proposal title correctly
    And I fill in proposal summary with text that meets requirements
    And I fill in proposal description with text that meets requirements
    And I try to proceed to next step
    Then I am presented with Open Proposal proposal lead stage
```

Then I would after that `When` step, that UI has no error states.
We can argue that "And I try to proceed to the next step" handles the verification.

My intuition here is to treat the Steps, like actually going through the UI as a user.
If I fill out a field "correctly", I also as a side effect check, that the field is not red.

---
Anyway, fine with leaving as is, and add more to it later.

Reply:
I think it has to stay like that for now, but later on if there is a better way lets improve it.
So at the moment "I try to proceed to next step" is only about clicking the proceed button.
The "I am presented with Open Proposal proposal lead stage" step is checking if the user is presented with the next stage and that's how we assume that the proposal stage passed the validation. 
That's true we could and probably should check if there are actual errors display, but we could only do that after the "proceed" button is clicked as that is when the validation happens (at least at the moment)